src/overlays/actors/ovl_En_Hgo/z_en_hgo.c

```diff
@@ -22,54 +52,343 @@ const ActorInit En_Hgo_InitVars = {
+                    break;
+            }
+        } else if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            switch (this->unk_218) {
+                case 1:
+                    if ((Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) && (this->unk_312 == 0)) {
+                        this->unk_312 = 1;
+                        if ((gSaveContext.sceneSetupIndex == 0) &&
+                            ((globalCtx->csCtx.unk_12 == 2) || (globalCtx->csCtx.unk_12 == 4))) {
+                            Audio_PlayActorSound2(&this->actor, NA_SE_VO_GBVO02);
+                        }
+                    }
+                    break;
+                case 2:
+                    this->unk_218 = 3;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 3);
+                    break;
+                case 5:
+                    this->unk_218 = 6;
+                    func_800BDC5C(&this->skelAnime, sAnimations, 6);
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hgo/func_80BD04E0.s")
+        func_800EDF24(&this->actor, globalCtx, actionIndex);
+        return 1;
+    }
+    if ((globalCtx->csCtx.state == 0) && (((gSaveContext.weekEventReg[75]) & 0x20)) &&
+        (this->actionFunc == func_80BD0410)) {
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0xC6, this->actor.focus.pos.x, this->actor.focus.pos.y,
+                    this->actor.focus.pos.z, 7, 0, 0, 0x7F5A);
+        func_80BD0420(this);
+    }
+    this->unk_316 = 0x63;
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hgo/func_80BD064C.s")
+void func_80BD0B8C(EnHgo* this, GlobalContext* globalCtx) {
+    func_800E9250(globalCtx, &this->actor, &this->unk_300, &this->unk_306, this->actor.focus.pos);
+    if (this->unk_30E >= 3) {
+        this->unk_30E--;
+    } else if (this->unk_30E == 2) {
+        this->unk_30C = 1;
+        this->unk_30E = 1;
+    } else if (this->unk_30E == 1) {
+        this->unk_30C = 2;
+        this->unk_30E = 0;
+    } else {
+        this->unk_30C = 0;
+        this->unk_30E = 0x3C;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hgo/func_80BD0660.s")
+void EnHgo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnHgo* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hgo/func_80BD06FC.s")
+    this->actionFunc(this, globalCtx);
+    SkelAnime_Update(&this->skelAnime);
+    if (func_80BD0898(this, globalCtx)) {
+        func_800E8F08(&this->unk_300, &this->unk_306);
+    } else {
+        if (this->actionFunc != func_80BD0410) {
+            if (this->actionFunc != func_80BD0434) {
+                Collider_UpdateCylinder(&this->actor, &this->collider);
+                CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+                func_80BD0B8C(this, globalCtx);
+            }
+        }
+    }
```

Comment:
I think you could combine the else and ifs into an `else if`. Even if it works I'll leave it up to you to decide which way to leave it.